classes/controller/FrontController.php

```diff
@@ -1489,6 +1490,67 @@ public function getTemplateVarPage()
         return $page;
     }
 
+    public function getBreadcrumb()
+    {
+        $breadcrumb = [];
+        $context = Context::getContext();
+
+        $breadcrumb[] = [
+            'title' => $this->getTranslator()->trans('Home', [], 'Breadcrumb'),
+            'url'   => $context->link->getPageLink('index', true)
+        ];
+
+        if ($context->controller->php_self == 'category') {
+            foreach ($context->controller->category->getAllParents() as $category) {
+                if ($category->id_parent != 0 && !$category->is_root_category) {
+                    $breadcrumb[] = $this->getCategoryPath($category, $context);
+                }
+            }
+        } elseif ($context->controller->php_self == 'product') {
```

Comment:
Maybe instead of having an elseif per controller type, we could instead have a version of `getBreadcrumb` in each controller that gets the breadcrumb for each controller. This way there is no need to check `php_self`.